jaggery-modules/rxt/module/scripts/permissions/permissions.js

```diff
@@ -946,6 +946,36 @@ var permissions = {};
         }
         return authorized;
     };
+
+    permissions.SYSTEM_ANON_ROLE = 'SYSTEM/wso2.anonymous.role';
+    permissions.checkActionPermissionforPath = function(path,action,tenantId,username) {
+        var username = typeof username !== 'undefined' ?  username : null; // Default parameter set to null
+        var server = require('store').server;
+        var isAuthorized = false;
+
+        if ((!tenantId)) {
+            throw 'Unable to resolve permissions without the tenantId';
+        }
+        var userManager = server.userManager(tenantId);
+        var authorizer = userManager.authorizer;
+
+        try {
+            if ((!username) || (username === wso2AnonUsername())) {
+                if(log.isDebugEnabled()){
+                    log.debug('[permissions] username not provided to check ' + permission + '.The anon role will be used to check permissions');
+                }
+                isAuthorized = authorizer.isRoleAuthorized(permissions.SYSTEM_ANON_ROLE, path, action);
+            } else {
+                if(log.isDebugEnabled()) {
```

Comment:
Thanks @mushthaq33 , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/e95caa53ae6d2bec4d278537d014af0496cdfa7a)